src/components/ChangePasswordModal.tsx

```diff
@@ -0,0 +1,124 @@
+import PopUp from './PopUp'
+import { Button, ModalProps, Spinner } from 'react-bootstrap'
+import { useForm } from 'react-hook-form'
+import Input from './Input'
+import { useEffect } from 'react'
+
+interface IChangePasswordModal extends ModalProps {}
+
+export interface IChangePasswordModalForm {
+  password: string
+  newPassword: string
+  newPasswordConfirm: string
+}
+
+export default function ChangePasswordModal({
+  onSubmit,
+  onHide,
+  ...modalProps
+}: IChangePasswordModal) {
+  const {
+    register,
+    watch,
+    getFieldState,
+    handleSubmit,
+    reset,
+    formState: { errors, isSubmitting },
+  } = useForm({
+    mode: 'onSubmit',
+    defaultValues: {
+      password: '',
+      newPassword: '',
+      newPasswordConfirm: '',
+    },
+  })
+
+  useEffect(() => {
```

Comment:
This seems unnecessary since we call `reset()` any time the modal closes. Was there an edge case that this is handling or is this left over experimentation? (in which case lets remove)

Reply:
This was added to deal with the case where an incorrect password is entered (and the modal is dismissed): when the modal is opened back up, the form state wasn't reset.

Reply:
Ah, I missed that it dismisses the modal on error. Lets remove that, as well as this useEffect hook.

Reply:
The linear ticket needs to be edited then because it's in the AC.